app/javascript/components/lease_ups/LeaseUpApplicationsPage.js

```diff
@@ -1,39 +1,81 @@
 import React from 'react'
-import { flow } from 'lodash'
+import { flow, isEmpty } from 'lodash'
 
 import LeaseUpApplicationsTableContainer from './LeaseUpApplicationsTableContainer'
 import TableLayout from '../layouts/TableLayout'
-import mapProps from '~/utils/mapProps'
 import { mapListing, mapApplicationPreference } from '~/components/mappers/soqlToDomain'
 import { buildLeaseUpModel } from './leaseUpModel'
 import appPaths from '~/utils/appPaths'
+import apiService from '~/apiService'
+import EagerPagination from '~/utils/EagerPagination'
+import Context from './context'
+import mapProps from '~/utils/mapProps'
+
+class LeaseUpApplicationsPage extends React.Component {
+  state = { loading: false, applications: [], pages: 0 }
+
+  constructor (props) {
+    super(props)
+    this.eagerPagination = new EagerPagination(20, 100)
+  }
+
+  fetchApplications = async (page) => {
+    const response = await apiService.fetchLeaseUpApplications(this.props.listing.id, page)
+    return {
+      records: !isEmpty(response.records) ? response.records.map(flow(mapApplicationPreference, buildLeaseUpModel)) : [],
+      pages: response.pages
+    }
+  }
 
-const LeaseUpApplicationsPage = ({ listing, applications }) => {
-  const pageHeader = {
-    title: listing.name,
-    content: listing.building_street_address,
-    action: {
-      title: 'Export',
-      link: `/listings/${listing.id}/lease-ups/export`
-    },
-    breadcrumbs: [
-      {title: 'Lease Ups', link: appPaths.toLeaseUps()},
-      {title: listing.name, link: appPaths.toLeaseUpApplications(listing.id)}
-    ]
+  loadPage = (page) => {
+    const fetcher = p => this.fetchApplications(p)
+    this.setState({ loading: true, page: page })
+    this.eagerPagination.getPage(page, fetcher).then(({ records, pages }) => {
```

Comment:
I know I did the same thing in ApplicationsPage and you got the code from there, but I'm wondering if this might be more readable:
make loadPage `async` and 
```
const { records, pages } = await this.eagerPagination.getPage(page, fetcher) 
this.setState({ applications: records, loading: false, pages: pages })
```